frontend/src/components/worksheets/items/ActionButtons.js

```diff
@@ -44,18 +70,53 @@ class ActionButtons extends React.Component<{
                     </Button>
                 ) : null}
                 {!showBundleOperationButtons ? (
-                    <Button
-                        size='small'
-                        color='inherit'
-                        aria-label='Add New Upload'
-                        className={classes.uploadButton}
-                        disabled={!editPermission}
-                    >
-                        <label className={classes.uploadLabel} for='codalab-file-upload-input'>
+                    <span>
+                        <Button
+                            size='small'
+                            color='inherit'
+                            id='upload-button'
+                            aria-label='Add New Upload'
+                            onClick={onShowNewUpload}
```

Comment:
By adding the button to trigger uploads instead of the label, I think there's a possibility that you're undoing the fixes done in #1945. Can you make sure that the following things still work:
- uploads still work on chrome
- when I open the upload file dialog and exit it, then click "upload file" again, the dialog still shows up
- when I upload a file X, then click "upload file" again and select the same file X, the same file gets uploaded again